be/src/exec/workgroup/scan_task_queue.cpp

```diff
@@ -18,6 +19,23 @@ bool FifoScanTaskQueue::try_offer(ScanTask task) {
     return true;
 }
 
+/// PriorityScanTaskQueue.
+PriorityScanTaskQueue::PriorityScanTaskQueue(size_t max_elements) : _queue(max_elements) {}
+
+StatusOr<ScanTask> PriorityScanTaskQueue::take(int worker_id) {
```

Comment:
I remember old priority scan strategy is complex, this simple strategy is enough?

Reply:
This is the same as the old priority scan strategy.
1. `OlapScanNode::_compute_priority()` decreases the priority of a task, if its scan node has committed too many tasks. 
    The next PR will apply this logic for pipeline, since this PR doesn't modify the logic and just refactor codes.
2.  After taking 512 tasks from the queue, the priority of remaining tasks in the queue will be upgrade 2. 
    This logic is in `BlockingPriorityQueue`, and `PriorityScanTaskQueue` is just an encapsulation of it.